client/src/components/SearchBar.js

```diff
@@ -0,0 +1,115 @@
+import React, { useState, useContext, useEffect } from 'react';
+import { makeStyles, withStyles } from '@material-ui/core/styles';
+import {
+  TextField,
+  Grid,
+  Button,
+  Container,
+  Box,
+} from '@material-ui/core';
+import { Link as RouterLink } from 'react-router-dom';
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import SearchIcon from '@material-ui/icons/Search';
+import axios from 'axios';
+
+const CustomTextField = withStyles({
+  root: {
+    '& .MuiOutlinedInput-root': {
+      '& fieldset': {
+        border: '0',
+      },
+    },
+  },
+})(TextField);
+
+const useStyles = makeStyles((theme) => ({
+    root: {
+        '& > *': {
+        margin: theme.spacing(1),
+        },
+    },
+    btn: {
+      paddingTop: theme.spacing(1.5),
+      paddingBottom: theme.spacing(1.5),
+      borderRadius: '30px',
+      marginLeft: '20px'
+    },
+    text: {
+        width: '350px',
+    },
+    box:{
+      backgroundColor: '#fbfcff',
+    },
+    bar: {
+        display: "flex",
+        flexWrap: "nowrap"
+    },
+    buttonProgress: {
+      color: '#DF1B1B',
+      position: 'absolute',
+      top: '2%',
+      left: '35%',
+    }
+}));
+
+export default function BasicTextFields(props) {
+    const classes = useStyles();
+    const [allUsers, setAllUsers] = useState([]);
+    const [searchUser, setSearchUser] = useState('');
+
+    const getAllUsers = async () => {
```

Comment:
This logic is very similar if not the same as what I use in the FindNewFriendsModal, we could eventually abstract this away to context to avoid repetition!